### PR TITLE
hotfix(staging_apiserver_pkg_httplog): restore depth to log calls

### PR DESCRIPTION
**Pull request by AbdulBasitAlvi**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

This PR adds depth to logging which was removed when migrating to structured logging.

#### Which issue(s) this PR fixes:

Ref #102353

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go [hunk 1]
```diff
@@ -97,7 +97,7 @@ func WithLogging(handler http.Handler, pred StacktracePred) http.Handler {
 		req = req.WithContext(context.WithValue(ctx, respLoggerContextKey, rl))
 
 		if klog.V(3).Enabled() {
-			defer func() { klog.InfoS("HTTP", rl.LogArgs()...) }()
+			defer func() { klog.InfoSDepth(1, "HTTP", rl.LogArgs()...) }()
```
Comment (serathius) [inline staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go:100]: Without InfoSDepth we needed to have the log call in the same function. Using InfoSDepth allows us to move it back to to dedicated function. 
PTAL https://github.com/kubernetes/kubernetes/pull/91833/files

Comment (AbdulBasitAlvi) [reply staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go]: @serathius I moved it into a dedicated function using a similar implementation used before, can you check now? 

Let me know if you have any further suggestions ( I am fairly new to this :P )

File: staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go [hunk 2]
```diff
@@ -180,38 +178,31 @@ func AddInfof(ctx context.Context, format string, data ...interface{}) {
...
-	args := []interface{}{
-		"verb", rl.req.Method,
-		"URI", rl.req.RequestURI,
-		"latency", latency,
-		"userAgent", rl.req.UserAgent(),
-		"audit-ID", auditID,
-		"srcIP", rl.req.RemoteAddr,
-		"resp", rl.status,
-	}
-	if len(rl.statusStack) > 0 {
```
Comment (serathius) [inline staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go:206]: What about statusStack and addedInfo args?

Comment (AbdulBasitAlvi) [reply staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go:206]: Thanks @serathius for pointing it out. I have modified the logging function to include these args as well.

File: staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go [hunk 3]
```diff
@@ -180,38 +180,33 @@ func AddInfof(ctx context.Context, format string, data ...interface{}) {
...
+			keysAndValues = append(keysAndValues, "statusStack", rl.statusStack)
+		}
+		info := rl.addedInfo.String()
+		if len(info) > 0 {
+			keysAndValues = append(keysAndValues, "addedInfo", info)
+		}
 	}
-	return args
+
+	klog.InfoSDepth(1, "HTTP", keysAndValues)
```
Comment (serathius) [inline staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go:209]:
```suggestion
	klog.InfoSDepth(1, "HTTP", keysAndValues...)
```

Comment (AbdulBasitAlvi) [reply staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go]: Thanks again! 👍

**Comment by serathius** [comment]
Can you please compare results from this change? 
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#verify-log-output shows how to do this.

**Comment by AbdulBasitAlvi** [comment]
> Can you please compare results from this change?
> https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#verify-log-output shows how to do this.

```
I0604 18:04:51.768758   63669 main.go:59] "HTTP" verb="GET" URI="/metrics" latency="1s" userAgent="Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0. 2272.118 Safari/537.36." audit-ID="" srcIP="127.0.0.1"
I0604 18:04:51.768850   63669 main.go:62] "HTTP" verb="GET" URI="/metrics" latency="1s" userAgent="Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0. 2272.118 Safari/537.36." audit-ID="" srcIP="127.0.0.1"

```

One question @serathius , should the string be "HTTP" or "Received HTTP request"? Otherwise everything else is the same

**Pull request merged**